daemon/container_linux.go

```diff
@@ -306,6 +307,24 @@ func populateCommand(c *Container, env []string) error {
 	return nil
 }
 
+func mergeDevices(defaultDevices, userSpecifiedDevices []*configs.Device) []*configs.Device {
+	if len(userSpecifiedDevices) == 0 {
+		return defaultDevices
+	}
+
+	// Only add those default devices, which are not specified by the user.
+	// Otherwise, the user specified devices end up without permissions.
+	var devs []*configs.Device
+	for _, defaultDevice := range defaultDevices {
+		for _, userSpecified := range userSpecifiedDevices {
+			if defaultDevice.Path != userSpecified.Path {
```

Comment:
Yes, this would be wrong.
But I think if we take the last line of `mergeDevices` into account:
```return append(devs, userSpecifiedDevices...)```
it does what I tried to describe.